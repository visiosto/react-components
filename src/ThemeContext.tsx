// Copyright (c) 2021 Visiosto oy
// Licensed under the MIT License

// Thanks to Joshua Comeau for the original code, licensed under MIT License:
// https://github.com/joshwcomeau/dark-mode-minimal

import { createContext } from 'react';

// eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
const defaultContext = { colorMode: 'light', setColorMode: (value: string) => {} };

export default createContext(defaultContext);
