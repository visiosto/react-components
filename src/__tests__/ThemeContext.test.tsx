// Copyright (c) 2021 Visiosto oy
// Licensed under the MIT License

import '../../test/matchMedia';

import React, { useContext } from 'react';

import ThemeContext from '../ThemeContext';

import renderWithProviders from '../../test/renderWithProviders';

function Component() {
  const { colorMode } = useContext(ThemeContext);

  return <div>{colorMode}</div>;
}

describe('Theme context component', () => {
  it('renders correctly', () => {
    const { container } = renderWithProviders(<Component />);

    expect(container).toMatchSnapshot();
  });
});
