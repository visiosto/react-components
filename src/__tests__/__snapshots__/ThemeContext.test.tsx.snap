// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme context component renders correctly 1`] = `
<div>
  <div />
</div>
`;
